src/middlewared/middlewared/plugins/vm/vms.py

```diff
@@ -216,6 +221,24 @@ async def common_validation(self, verrors, schema_name, data, old=None):
                 'Invalid bootloader ovmf choice specified'
             )
 
+        if data['bootloader'] == 'UEFI' and not data['nvram_location']:
+            verrors.add(
+                f'{schema_name}.nvram_location',
+                'NVRAM location must be specified when booting with UEFI'
+            )
+
+        if data['nvram_location'] and await self.middleware.run_in_thread(
```

Comment:
What if the path does not exist but its parent is not writeable by libvirt user so automatic creation will fail?

Reply:
I will add the validation ✅